app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -1382,103 +1379,36 @@ public void onDialogDismissed(int dialogId) {
-            datePickerDialog.getDatePicker().setMinDate(minDate != null ? minDate.getTime() : getDefaultMinDateOfDatePicker());
-            datePickerDialog.getDatePicker().setMaxDate(maxDate != null ? maxDate.getTime() : getDefaultMaxDateOfDatePicker());
-            return datePickerDialog;
-        }
-
-        private long getDefaultMinDateOfDatePicker() {
-            Calendar minDateCalendar = Calendar.getInstance();
-            minDateCalendar.set(1970, 0, 1);
-            return minDateCalendar.getTimeInMillis();
-        }
-
-        private long getDefaultMaxDateOfDatePicker() {
-            Calendar maxDateCalendar = Calendar.getInstance();
-            maxDateCalendar.set(2100, 11, 31);
-            return maxDateCalendar.getTimeInMillis();
-        }
-
-        public void onDateSet(DatePicker view, int year, int month, int day) {
-            Calendar c = new GregorianCalendar();
-            c.set(Calendar.YEAR, year);
-            c.set(Calendar.MONTH, month);
-            c.set(Calendar.DAY_OF_MONTH, day);
-            c.set(Calendar.HOUR_OF_DAY, 0);
-            c.set(Calendar.MINUTE, 0);
-            c.set(Calendar.SECOND, 0);
-            c.set(Calendar.MILLISECOND, 0);
-
-            long unixTime = c.getTimeInMillis();
+    private void showDatePicker(
+            @Nullable Date minDate,
+            @Nullable Date maxDate,
+            MaterialPickerOnPositiveButtonClickListener<Long> listener
+    ) {
+        // Create a new instance of MaterialDatePicker and return it
+        CalendarConstraints calendarConstraints = new CalendarConstraints.Builder()
+                .setStart(minDate != null ? minDate.getTime() : getDefaultMinDateOfDatePicker())
+                .setEnd(maxDate != null ? maxDate.getTime() : getDefaultMaxDateOfDatePicker())
+                .build();
+
+        MaterialDatePicker<Long> materialDatePicker = MaterialDatePicker.Builder.datePicker()
+                .setSelection(Calendar.getInstance().getTimeInMillis())
+                .setCalendarConstraints(calendarConstraints)
+                .build();
+
+        materialDatePicker.addOnPositiveButtonClickListener(listener);
+        materialDatePicker.show(getSupportFragmentManager(), TAG);
+    }
 
-            Date date = new Date(unixTime);
+    private long getDefaultMinDateOfDatePicker() {
+        Calendar minDateCalendar = Calendar.getInstance();
+        minDateCalendar.set(1970, 0, 1);
+        return minDateCalendar.getTimeInMillis();
+    }
 
-            Bundle result = new Bundle();
-            result.putSerializable(TEXT_FIELD_TO_EDIT_ARGUMENT_KEY, textFieldEdit);
-            result.putSerializable(NEWLY_PICKED_DATE_ARGUMENT_KEY, date);
-            getParentFragmentManager().setFragmentResult(PICK_DATE_REQUEST_KEY, result);
-        }
+    private long getDefaultMaxDateOfDatePicker() {
+        Calendar maxDateCalendar = Calendar.getInstance();
+        maxDateCalendar.set(2100, 11, 31);
```

Comment:
This seems really arbitrarily chosen. If we really need this, I would recommend at least putting it in a constant instead of hardcoding it here. Same for the min date, though that date is less arbitrary :)

Reply:
Also not new code: https://github.com/CatimaLoyalty/Android/commit/3a9b92231eb8dcb1aa43d089e1a3108c9ebd061d

Reply:
Yeah, just wondering if it can be improved.

Reply:
This was added in https://github.com/CatimaLoyalty/Android/pull/1083#discussion_r996450358, it is the same as the Android default. It was added because "it makes it more explicit" but I guess it can be removed.

Reply:
Done